starlette_context/middleware/mixin.py

```diff
@@ -0,0 +1,44 @@
+import logging
+from typing import Callable, Optional
+
+from starlette_context.errors import StarletteContextException
+from starlette.responses import PlainTextResponse
+
+
+class StarletteContextMiddlewareMixin:
+    _default_error_handler = lambda self, e: PlainTextResponse(  # noqa: E731
+        content=e.detail, status_code=e.status_code
+    )
+
+    def __init__(
+        self,
+        error_handler: Optional[Callable] = None,
+        log_errors: bool = False,
+        *args,
+        **kwargs,
+    ):
+        self.error_handler = error_handler or self._default_error_handler
+        self.log_errors = log_errors
+
+    def log_error(self):
+        if self.log_errors:
+            logger = self.get_logger()
+            logger.exception("starlette-context exception")
+
+    @staticmethod
+    def get_logger():
+        return logging.getLogger("starlette_context")  # pragma: no cover
+
+    def create_response_from_exception(
+        self, e: StarletteContextException
+    ) -> PlainTextResponse:
```

Comment:
Since it can return any kind of response, `Response` would be the proper return type